### PR TITLE
Update reporting record limit tip

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
There's an outdated tip on the reporting page related to https://github.com/opensearch-project/dashboards-reporting/issues/328 that we weren't aware existed when the feature was implemented. This PR updates the tip: there's no longer a hardcoded 10k record limit.

### Issues Resolved
N/A

### Version
2.17 and above.

### Frontend features
A new "Record limit" field is available, which shows a notice if the limit is configured above the default.
<p>
image
image
</p>

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Swiddis** [comment]
@kolchfa-aws at your convenience, I know things are hectic right now with 2.18.

File: _reporting/report-dashboard-index.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ redirect_from:
 
 You can use OpenSearch Dashboards to create PNG, PDF, and CSV reports. To create reports, you must have the correct permissions. For a summary of the predefined roles and the permissions they grant, see the [Security plugin]({{site.url}}{{site.baseurl}}/security/access-control/users-roles#predefined-roles).
 
-CSV reports have a non-configurable 10,000 row limit. They have no explicit size limit (for example, MB), but extremely large documents could cause report generation to fail with an out of memory error from the V8 JavaScript engine.
+CSV reports have a non-configurable 10,000 row limit up to version 2.16. As of version 2.17, this limit can be configured when setting up a report. While reports have no explicit size limit (for example, MB), extremely large documents could cause report generation to fail with an out of memory error from the V8 JavaScript engine.
```
Comment (kolchfa-aws) [inline _reporting/report-dashboard-index.md:14]:
```suggestion
CSV reports have a non-configurable 10,000 row limit up to OpenSearch version 2.16. As of version 2.17, this limit can be configured when setting up a report. While reports have no explicit size limit (for example, MB), extremely large documents could cause report generation to fail with an out of memory error from the V8 JavaScript engine.
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @Swiddis! Just one suggestion.

File: _reporting/report-dashboard-index.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ redirect_from:
 
 You can use OpenSearch Dashboards to create PNG, PDF, and CSV reports. To create reports, you must have the correct permissions. For a summary of the predefined roles and the permissions they grant, see the [Security plugin]({{site.url}}{{site.baseurl}}/security/access-control/users-roles#predefined-roles).
 
-CSV reports have a non-configurable 10,000 row limit. They have no explicit size limit (for example, MB), but extremely large documents could cause report generation to fail with an out of memory error from the V8 JavaScript engine.
+CSV reports have a non-configurable 10,000 row limit up to OpenSearch version 2.16. As of version 2.17, this limit can be configured when setting up a report. While reports have no explicit size limit (for example, MB), extremely large documents could cause report generation to fail with an out of memory error from the V8 JavaScript engine.
```
Comment (natebower) [inline _reporting/report-dashboard-index.md:14]:
```suggestion
CSV reports have a non-configurable 10,000-row limit in OpenSearch version 2.16 and earlier. As of version 2.17, this limit can be configured when setting up a report. While reports have no explicit size limit (for example, MB), extremely large documents could cause report generation to fail with an out-of-memory error from the V8 JavaScript engine.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @Swiddis Just a few changes. Thanks!

**Pull request merged**